shm/inc/delegate.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <cstddef>
```

Comment:
Here the same question.

Reply:
Maybe some compilers do not need a header to recognise 'size_t' (similarly when using 'string' my compiler does not make me to include <string> header).